From b4d9ad138cda3d7a3095a3ab0d55d57bded5453b Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Mon, 23 May 2022 12:47:43 +0530
Subject: [PATCH] ath12k: Add coredump segment for dynamic paging

Add a separate segment in the coredump for collecting dynamic paging
bin as a part of ramdump version v2.

Note: Add other new dump as a separate segment to follow ramdump
version v2.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/coredump.c | 10 +++++++++-
 drivers/net/wireless/ath/ath12k/coredump.h |  2 ++
 2 files changed, 11 insertions(+), 1 deletion(-)

Index: backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/coredump.c
===================================================================
--- backports-20220404-5.4.164-f40abb4788.orig/drivers/net/wireless/ath/ath12k/coredump.c	2022-05-31 17:03:20.094883501 +0530
+++ backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/coredump.c	2022-05-31 17:04:07.182498586 +0530
@@ -115,7 +115,7 @@
 	strscpy(file_data->df_magic, "ATH12K-FW-DUMP",
 		sizeof(file_data->df_magic));
 	file_data->len = cpu_to_le32(header_size);
-	file_data->version = cpu_to_le32(ATH12K_FW_CRASH_DUMP_VERSION);
+	file_data->version = cpu_to_le32(ATH12K_FW_CRASH_DUMP_V2);
 	file_data->chip_id = cpu_to_le32(ar_pci->dev_id);
 	file_data->qrtr_id = cpu_to_le32(ar_pci->ab->qmi.service_ins_id);
 	file_data->bus_id = pci_domain_nr(ar_pci->pdev->bus);
@@ -208,10 +208,18 @@
 		if (ab->qmi.target_mem[i].type == CALDB_MEM_REGION_TYPE) {
 			if (ath12k_cold_boot_cal ||
 			    ab->hw_params->cold_boot_calib) {
+				/* Add separate segment type for CALDB*/
 				skip_count++;
 				continue;
 			}
 		}
+		if (ab->qmi.target_mem[i].type == PAGEABLE_MEM_REGION_TYPE) {
+			seg_info->len = ab->qmi.target_mem[i].size;
+			seg_info->addr = ab->qmi.target_mem[i].paddr;
+			seg_info->vaddr = ab->qmi.target_mem[i].v.ioaddr;
+			seg_info->type = ATH12K_FW_PAGEABLE_DATA;
+			seg_info++;
+		}
 		seg_info->len = ab->qmi.target_mem[i].size;
 		seg_info->addr = ab->qmi.target_mem[i].paddr;
 		seg_info->vaddr = ab->qmi.target_mem[i].v.addr;
Index: backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/coredump.h
===================================================================
--- backports-20220404-5.4.164-f40abb4788.orig/drivers/net/wireless/ath/ath12k/coredump.h	2022-05-31 17:03:20.094883501 +0530
+++ backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/coredump.h	2022-05-31 17:03:20.090883534 +0530
@@ -7,12 +7,18 @@
 #define _COREDUMP_H_
 
 #define ATH12K_FW_CRASH_DUMP_VERSION 1
+#define ATH12K_FW_CRASH_DUMP_V2      2
 
 enum ath12k_fw_crash_dump_type {
 	ATH12K_FW_CRASH_PAGING_DATA,
 	ATH12K_FW_CRASH_RDDM_DATA,
 	ATH12K_FW_REMOTE_MEM_DATA,
+	ATH12K_FW_PAGEABLE_DATA,
+	ATH12K_FW_M3_DUMP,
 	ATH12K_FW_QDSS_DATA,
+	ATH12K_FW_CALDB,
+        ATH12K_FW_AFC,
+        ATH12K_FW_MLO_GLOBAL,
 };
 
 struct ath12k_dump_segment {
